File: sort/insertion/insertion-sort.py
def insertion_sort(arr) -> list:
    for index in range(1,len(arr)):
        swap = True
        ind = index
        s_index= index-1
        while swap == True and s_index >= 0:
            if arr[ind] < arr[s_index]:
                temp = arr[ind]
                arr[ind] = arr[s_index]
                arr[s_index] = temp
                ind -=1
                s_index -=1
            else:
                swap = False
    return arr

if __name__=="__main__":
    array = []
    while True:
        temp = input("Enter a value into the array (press enter to finish): \n")
        try:
            if temp:
                array.append(int(temp))
            else:
                break
        except:
            print("integers only please")

    print(insertion_sort(array) if array else "Please give a valid entry")

File: sort/insertion/insertionSort.go
package main

func insertion_sort(arr []int) []int {
	if len(arr) <= 0 {
		return []int{}
	}

	for index := 1; index < len(arr); index++ {
	out:
		for temp_index, second_index := index, index-1; second_index > 0; temp_index, second_index = temp_index-1, second_index-1 {
			var left int = arr[second_index]
			var right int = arr[temp_index]
			if left > right {
				arr[second_index], arr[temp_index] = right, left
			} else {
				break out
			}
		}
	}
	return arr

}
